### DISCUSSION TITLE
Recurring Configuration Check, Errors

### DISCUSSION THREAD

**Discussion by arrmo** [tagyoureit/nodejs-poolController#640] [App and O/S Config Questions] [answered]
Hi,

This may all be fine, just not sure 😆. I see this in my logs, seems like about every 20 minutes? Not sure why the configuration needs to be rechecked every 20 min ... LOL. But also, are the errors normal / expected?

Thanks!

```
1|njsPC  | [12/2/2022, 8:56:01 AM] info: Requesting intellitouch configuration
1|njsPC  | RS485 Stats:{ "bytesReceived": 1320732 "success": 31560, "failed": 310, "bytesSent": 83795, "collisions": 69, "failureRate": 0.97% }
1|njsPC  | RS485 Stats:{ "bytesReceived": 1320761 "success": 31560, "failed": 311, "bytesSent": 83807, "collisions": 69, "failureRate": 0.98% }
1|njsPC  | RS485 Stats:{ "bytesReceived": 1321054 "success": 31570, "failed": 312, "bytesSent": 83927, "collisions": 69, "failureRate": 0.98% }
1|njsPC  | RS485 Stats:{ "bytesReceived": 1321105 "success": 31570, "failed": 313, "bytesSent": 83951, "collisions": 69, "failureRate": 0.98% }
1|njsPC  | RS485 Stats:{ "bytesReceived": 1321616 "success": 31585, "failed": 314, "bytesSent": 84131, "collisions": 69, "failureRate": 0.98% }
1|njsPC  | RS485 Stats:{ "bytesReceived": 1322038 "success": 31600, "failed": 315, "bytesSent": 84311, "collisions": 69, "failureRate": 0.99% }
1|njsPC  | RS485 Stats:{ "bytesReceived": 1322483 "success": 31615, "failed": 316, "bytesSent": 84491, "collisions": 69, "failureRate": 0.99% }
1|njsPC  | RS485 Stats:{ "bytesReceived": 1322930 "success": 31630, "failed": 317, "bytesSent": 84671, "collisions": 69, "failureRate": 0.99% }
1|njsPC  | RS485 Stats:{ "bytesReceived": 1323383 "success": 31645, "failed": 318, "bytesSent": 84851, "collisions": 69, "failureRate": 0.99% }
1|njsPC  | RS485 Stats:{ "bytesReceived": 1323840 "success": 31660, "failed": 319, "bytesSent": 85031, "collisions": 69, "failureRate": 1.00% }
1|njsPC  | [12/2/2022, 8:56:47 AM] warn: Message aborted after 4 attempt(s): 165,1,16,33,214,1,0,1,174
1|njsPC  | [12/2/2022, 8:56:51 AM] warn: Message aborted after 4 attempt(s): 165,1,16,33,216,1,1,1,177
1|njsPC  | [12/2/2022, 8:56:56 AM] warn: Message aborted after 4 attempt(s): 165,1,16,33,216,1,2,1,178
1|njsPC  | [12/2/2022, 8:57:00 AM] warn: Message aborted after 4 attempt(s): 165,1,16,33,216,1,3,1,179
1|njsPC  | [12/2/2022, 8:57:04 AM] warn: Message aborted after 4 attempt(s): 165,1,16,33,216,1,4,1,180
1|njsPC  | [12/2/2022, 8:57:09 AM] warn: Message aborted after 4 attempt(s): 165,1,16,33,216,1,5,1,181
1|njsPC  | RS485 Stats:{ "bytesReceived": 1326102 "success": 31730, "failed": 320, "bytesSent": 85955, "collisions": 69, "failureRate": 1.00% }
1|njsPC  | [12/2/2022, 8:57:13 AM] warn: Message aborted after 4 attempt(s): 165,1,16,33,216,1,6,1,182
1|njsPC  | [12/2/2022, 8:57:17 AM] warn: Message aborted after 4 attempt(s): 165,1,16,33,216,1,7,1,183
1|njsPC  | [12/2/2022, 8:57:21 AM] warn: Message aborted after 4 attempt(s): 165,1,16,33,216,1,8,1,184
1|njsPC  | [12/2/2022, 8:57:23 AM] info: EasyTouch system config complete.
1|njsPC  | [12/2/2022, 8:57:23 AM] info: Initializing Nixie Controller
1|njsPC  | [12/2/2022, 8:57:23 AM] info: Nixie Controller Initialized
```

**Comment by rstrouse** [comment]
Yes these are normal.  It is just probing for features that are not returned by your panel.

**Comment by arrmo** [comment]
OK, makes sense - was just surprised to see it every 20 min or so ... LOL.

And the RS485 errors, also normal?

Thanks!

**Comment by rstrouse** [comment, marked as answer]
There will always be some failure rate on the RS485 bus.  For EasyTouch and IntelliTouch njsPC checks your system periodically for changes.  Normally an external controller like MobileTouch would do this when it wakes up but njsPC doesn't sleep.  For indoor control panels they have this manual thingy to synchronize the panel from the OCP in IntelliTouch.  njsPC is just eliminating that since it can simply ask for this info periodically, and suspend the processing when the bus gets busy.

I don't know if 20 minutes is the right interval or not but doing this makes sure we have parity with the OCP.  IntelliCenter does this differently and the probe simply asks for the current version of each configuration category.  If nothing has changed on the master we don't even ask.  Bear in mind the RS485 bus on IntelliCenter is at least twice as busy.

**Reply by tagyoureit** [reply]
20 minutes is what the ScreenLogic does, so that's where that number came from. I imagine nobody ever is changing their config that much (other than when there are random equipment changes or a new install) but there is also zero harm in requesting it.

**Reply by arrmo** [reply]
Makes sense, thanks! Was asking because I hit it a few times 😆. From above ... down about 1.5 min for the check, out of 20 min => ~ 10% down time.

Thanks again.

**Reply by tagyoureit** [reply]
If you had a raspberry pi directly wired to the rs485 bus it would be a lot faster. Just saying.

**Reply by arrmo** [reply]
> If you had a raspberry pi directly wired to the rs485 bus it would be a lot faster. Just saying.

LOL - get your point 😆. But - I have a (much faster) laptop directly connected to RS-485, so shouldn't be worse - right?

Thanks!

**Reply by rstrouse** [reply]
Wait did you say it was down when it was doing the check?  That check does not take the system down as it is lazy and adds requests to the queue without stopping others from being added.  It is simply loading and all other functions should do their thing without interruption.  There will be an indication in dashPanel that it is doing this.

**Reply by arrmo** [reply]
> Wait did you say it was down when it was doing the check? That check does not take the system down as it is lazy and adds requests to the queue without stopping others from being added. It is simply loading and all other functions should do their thing without interruption. There will be an indication in dashPanel that it is doing this.

You are right! And sorry, bad wording on my part. I did think (initially) it was down, but it's not ... just not so responsive (i.e. slowly handles the actions) - which makes sense, given your comments.

Thanks!
